### PR TITLE
fix ConvertToNamedArg for constructor invocations

### PR DESCRIPTION
**Pull request by jkciesluk**
fixes https://github.com/scalameta/metals/issues/4276

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala [hunk 1]
```diff
@@ -18,19 +18,27 @@ class ConvertToNamedArguments(trees: Trees) extends CodeAction {
...
-          case (arg, index)
-              if !arg.isInstanceOf[Term.Assign] && !arg
-                .isInstanceOf[Term.Block] =>
-            index
-        }
-        if (argIndices.isEmpty) firstApplyWithUnnamedArgs(apply.parent)
-        else Some(ApplyTermWithArgIndices(apply, argIndices))
+        getTermWithArgs(apply, apply.args)
+      case Some(newAppl @ Term.New(init)) =>
+        getTermWithArgs(newAppl, init.argss.flatten)
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala:41]: It doesn't work if the constructor has multiple parameters

```scala
class Foo(x: Int, y: Int)(z: Int)

new Foo(1, 2)(3)

// If we run the code action for `Foo(...)`
new Foo(1, 2)(z = 3)
```

Maybe we should treat differently for `Term.New`.

**Review by tanishiking — CHANGES_REQUESTED** [review]
I found a corner case bug, maybe the code action should behave differently for `Term.New`.
but overall, looks nice :+1:

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala (hunk 1 above)
Comment (jkciesluk) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala:41]: I've encountered a lot of problems while working on case when the constructor has multiple parameter lists,
from finding which parameter list to extract, to getting correct type information (in scala I get things like `dotty.tools.dotc.core.Types$UnspecifiedErrorType$@6a25de0a`, in scala2 I get nothing :sweat_smile:), so for now this code actions will not be showing for `new` with multiple parameter lists

Comment (jkciesluk) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertToNamedArguments.scala:41]: There were some issues with finding which parameter list is selected (in cases like `Foo(1,2)<<(>>3)`) so, at least for now, we are converting all parameters for `Term.New`

File: tests/unit/src/test/scala/tests/codeactions/ConvertToNamedArgumentsLspSuite.scala [hunk 2]
```diff
@@ -198,4 +236,12 @@ class ConvertToNamedArgumentsLspSuite
...
     filterAction = filterAction,
   )
+
+//   checkNoAction(
+//     "new-multiple-lists",
+//     """|object Something {
+//        |  class Foo(param1: Int)(param2: Int)
+//        |  val a = new Foo(<<1>>)(2)
+//        |}""".stripMargin,
+//   )
```
Comment (tanishiking) [inline tests/unit/src/test/scala/tests/codeactions/ConvertToNamedArgumentsLspSuite.scala:246]:
```suggestion

```

**Review by tanishiking — APPROVED** [review]
I left one nitpick comment, but LGTM! good job :tada:

**Comment by tgodzik** [comment]
@jkciesluk seems `ConvertToNamedArgumentsLspSuite` are failing, could you take a look?

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**